Soal_2/src/App.js

```diff
@@ -0,0 +1,53 @@
+import { Component } from "react";
+// import animeList from "./dummy-data";
+import MovieCard from "./MovieCard";
+
+
+class App extends Component {
+  constructor(props){
+    super(props)
+    this.state = {
+      movieList: [],
+      isLoading: true
+    }
+  }
+
+  componentDidMount(){
+    setTimeout(() => this.fetchMovieList(), 1000);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.movieList.length !== this.state.movieList.length) {
+      this.setState({isLoading: false});
+    }
+  }
+
+  fetchMovieList(){
+    fetch("https://api.jikan.moe/v3/top/anime")
+      .then(response => response.json())
+      .then(data => console.log(this.setState({movieList: data?.top})))
```

Comment:
ini kamu ngapain `console.log` ya dev ? 

padahal kan `this.setState({movieList: data?.top})` saja sudah cukup dev